From 117c045334ad694165ce0a0a5b054ab160257f41 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 25 Jul 2013 11:21:05 +0800
Subject: [PATCH 691/691] ENGR00272558 fastboot_dev and bootcmd cannot be
 configured for i.MX6 SabreSD

Only init the fastboot_dev and bootcmd env in the first boot

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 board/freescale/mx6q_sabresd/mx6q_sabresd.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/board/freescale/mx6q_sabresd/mx6q_sabresd.c b/board/freescale/mx6q_sabresd/mx6q_sabresd.c
index 8caca4a..6f866aa 100644
--- a/board/freescale/mx6q_sabresd/mx6q_sabresd.c
+++ b/board/freescale/mx6q_sabresd/mx6q_sabresd.c
@@ -1778,12 +1778,16 @@ int board_late_init(void)
 #ifdef MX6Q_SABRESD_ANDROID_H
 	switch (get_boot_device()) {
 	case SD_BOOT:
-		setenv("fastboot_dev", "mmc2");
-		setenv("bootcmd", "booti mmc2");
+		if (!getenv("fastboot_dev"))
+			setenv("fastboot_dev", "mmc2");
+		if (!getenv("bootcmd"))
+			setenv("bootcmd", "booti mmc2");
 		break;
 	case MMC_BOOT:
-		setenv("fastboot_dev", "mmc3");
-		setenv("bootcmd", "booti mmc3");
+		if (!getenv("fastboot_dev"))
+			setenv("fastboot_dev", "mmc3");
+		if (!getenv("bootcmd"))
+			setenv("bootcmd", "booti mmc3");
 		break;
 	default:
 		printf("unsupported boot devices\n");
-- 
1.7.7.4

